### ISSUE TITLE
Post-method docstring support

### ISSUE THREAD

**Issue by omus** [tonyhffong/Lint.jl#125]
Lint doesn't appear to handle docstring when they are made after the method declaration:

```julia
julia> lintstr("""
func(v) = v

\"\"\"
Documentation
\"\"\"
func(v)
""")
1-element Array{Lint.LintMessage,1}:
 none:3 E321 v: use of undeclared symbol
```

**Comment by tonyhffong** [comment]
Docstring isn't required to trigger this issue. This is sufficient.
```
julia> lintstr( "func(v)" )
1-element Array{Lint.LintMessage,1}:
 none:1 E321 v: use of undeclared symbol
```

**Comment by tonyhffong** [comment]
So basically lint thinks ```func(v)``` is a function call to ```func``` using an argument ```v```

**Comment by tonyhffong** [comment]
but we do want to throw a lint error in this case
```julia
function f(v)
    """
    some string
    """
    g(x) # x is not declared
end
```
So we want to grandfather undeclared variable when encountered at the "top-level" parse tree, AND that it was immediately preceded by a literal string. A bit tricky but probably doable.

**Comment by TotalVerb** [comment]
@tonyhffong: Lint is correct that `func(v)` is a function call to `func` using argument `v`. Only with a docstring should this be interpreted differently.

**Comment by MichaelHatherly** [comment]
Would it not be sufficient to check for `:call` expressions in the `@doc` linting [code](https://github.com/tonyhffong/Lint.jl/blob/a16c6cb8d63fa3360dd9a72510c9e2be6f3c81da/src/macros.jl#L62-L77) and skip them when encountered since

```julia
"..."
func(x)
```
is parsed as `:(@doc "..." func(x))`. There doesn't seem to be a need to do anything too complex here, am I missing something obvious? I'm not all that familiar with the Lint codebase.

Note that there's also the multidoc syntax

```julia
"..."
a, b, c
```

where `a`, `b`, and `c` can themselves be any expression, potentially call expressions with undefined variables.

**Comment by tonyhffong** [comment]
aha, obviously I haven't played with the parser for a while as I missed this. Yes that should lead to a solution.

**Comment by tonyhffong** [comment]
I think I have a fix in place.
